### DISCUSSION TITLE
[Application Development] <Any diagram for Fault proof (smart contracts)?>

### DISCUSSION THREAD

**Discussion by DeltaXV** [ethereum-optimism/developers#530] [Application Development]
### Did you check the documentation?

- [ ] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

Hi guys, 

Would like to know if there was any diagram for the faultproof (smart contracts architecture)?

ty

**Comment by imtipi** [comment]
https://specs.optimism.io/fault-proof/index.html

**Comment by joohhnnn** [comment]
[Before FP](https://mermaid.live/view#pako:eNq1V99zmzgQ_lcY3Ys7g41xnDjQTmfa1NPejHvOxHcvN36RYW1rAohKInEa53-_NRgsAY6bm5oHBkn77X7aXxLPJOAhEJ-sBE3X1uRunlj4yGxRTMzJeKNAJDSybniiBA2UnJNCaPeUq-O7m0G_U8nmwwPiXStgNHANBI4bEEjC4qPBa-Jas5gK1c5r4s4UTUIqws-ChSvo1Cfe6bKF8UpSHxpyN4JL-YXHlCXfQUpIViA67dM6bjDNVJqpKbKMUL8x1OSmqWIxk_EtF4pGHXOoyc2yFESwRmu49SVbdeoTuuyTVBCXctrgpH__kSCsPxMMD_qW8cRw725RdvL3SUWF1aOqbgVPOSrq3N5Nb6ez8Z3G_muG8WI06dx8-zSZjP_6Wq3WzAURC-4bQUfja6VS6TvOiql1tugFPHZArUFAFnf53sFO9aEEgBPCA0Q8dVIa3NMVSCcoE6y7gFDw4N6RInAmrlO315M8mpM8N9sW5slcHajqSXZ-oro1jWZz2iTZltnnJ9tmVSN9fNkgb5TZ2Ukb1g5kW6Z1kmaNn5ukaa0i2Tatk6y3l3PTrNuriLYvGFS1Dnd2mpqtA8XGpN6myl5ndbsft2kxsKSiCiye54nc1vK2wJV9MMcJiPkDWCx5oBELfwVevI2j2vqwUxWzRDmLTCS4d5rIJYhto4k2wLtj-hW03lJ0843mnOvA6g0dAQEw3FKM1Yw-3x5pPKUeo3P-by2trS3Xtg-9067VrJYje-v2UM-PDMSTldKsDPK2UUqtW3oruHUnb1NS60O_CxzkUlujLE8aLMpin9ivZnR-_yhiFgKCmHIesdJDQR-t8d_fnDzTj2X0q-A8HO3ZXOD2BfwADmwgyLACSzDuJi-H_VWnPWHKWwvFwMHSgo2qbrG4b8Hvwf9jufTs4rv7yEK19gfp5n0NGLlN3GKx_AXcePpJQyxOIjALBF7hDuw8ehIj87DnF8m3wALBFAvQjyWov-y_CjJbm13rVbpzDVQ9L2wz3HZ7Zdn1IrCNQ0eLiGGsSJu9F42Vz1QFeCbZ5eFgV93-4D8DYCaUXbvolN57T2wSg0DuIf7ePe8UzAkefjHMiY-fISxpFqndAfWCojRTfPaUBMRXIgObCJ6t1sRf0kjiKEtDLMYvjOKlPi5FUpr8y3lcCeGY-M9kQ3zXHfS8i9Gl6w3dwej6qj-yyRPxu67b8zxvcH3R719depfucPBik5-5DrfXv_aGQ290dTHwrvsXQ4RAyBQX34sf1Pw_9eU_ZU8STw)
[After FP](https://mermaid.live/view#pako:eNq9WNtu2zgQ_RWB--ICsmU5V6mLBdIkzRZIN0XSosDCL7Q4tolIopakkrhJ_n1Hki-kRCfNYt08BLzMmTkczoXyI0kEAxKTmaTF3Lu8Huce_qly0iyMyfmDBpnT1DsVuZY00WpMGqHqb7V7fn06GvbWsvV0g3jnBByNQguB8w4EclYPOrQuQ-8mo1K7aV2GN5rmjEr2QXI2g1574Z0p29heS5pTS-5UCqXOREZ5_hmUgnwGsudeNnBXheYZV9kXITVNe_bUkLspC5DJHLXgkaZ81msvmLILpSFbyRkTQ-aMq6LUcEEz-Ij-EXLR6y4Z8h9pmWpDotdeMHVDShfAvp9__bNnjFuX1r21bwqk9ynHO0fzXOTWpVWbqlf_f1VRc-atqi5KvGlO897Ft5Prs08nf7kVJilPbjvBgurnWhcqDoIZ1_NyMkhEFoCeg4Qy64vlBQbrgZYAAYM7SEURFDS5pTNQQbIKzP4EmBTJbaBkElyGQdveQIl0TOqYdm2M87HeUDWDc_dETWsGze6yTdKVEbsn67JqkN6-bZEfXZUaI_4KjaS797BlbUPWsWyStGvIrkna1tYkXcsmyXb52jXNtr01UfeGRdWooDunadjaUOwsmgWqW7Z3SpI15oKu2TXfbVsm63b7-CWc20bXjN0bVRCM9dLLmz72S6jeo77AMLpxbnsNWeadx5b3e7__x1PGcx1MSpkjBZqrKcinTj_rgKuH1gtos7qb3bLTJ2sdWEhZICEBfgdehoUVj_60pQes9FhN7D9rcXaZWtvyBgK3VrtwbTlbf4B6_ikBc62gpQJPaaoR3C4mziO9Few8yduUtFrC_wVOaqknq0K-hplVyU4Zk3iUJ0fx-in8km87bRusoyBW155IQJDyeI7oPMGRmG5T0alOdRSiKsCA0ZJjLnsTkbPqBJt8NNOhfqg28cagEIrr4B6LBZP03kPRoM7Sbdn4IrgOJXcmNrg6xqW4gwAeIMETeCswerJO5eWb2B3sGyWNx7xlSao9_8KVGcbxy4snvKjAPG_j3R5vau3qZV6rmaSYqClXuq2hzbqDxBDGJPEwxApINLAmaPSicOX3qitRTDKYevCg1x-NGGdS3EL823Qa-c24f8-Znsej4uF9C5iGXdxkMv0J3PnViYGYvIrAjJX4bbNhF9FXMapO0foL6y2wRHK8SoybFWg4Hb4IstuQ3-orpnMtVDsPfDu8fXcV9NsFy7feasaNWMaaSF160dr5QHWCbdz_IgWmHQ7WYbXxnwWwo8nv5obfDnfffEms3Pue-CQDiYdjJCaPlYUxwQcFZgeJcchgWqmpmv0zitJSi5tFnpBYyxJ8IkU5m5N4SlOFs7JgmHlnnOLncLYSKWj-txDZWgjnJH4kDyQOw9Eg2js6CKP9cHR0fDg88smCxP0wHERRNDreGw4PD6KDcH_07JMftY5wMDyO9vejo8O9UXQ83NtHCDCO5_3c_F5U_2z0_C-RhmRh)
